src/main/java/com/kodilla/ecommercee/domain/Cart.java

```diff
@@ -0,0 +1,21 @@
+package com.kodilla.ecommercee.domain;
+
+import com.kodilla.ecommercee.dto.ProductDto;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.Entity;
+import java.util.List;
+
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+@Entity
+public class Cart {
+
+    private Long id;
+    private Long userId;
+    private List<ProductDto> productList;
```

Comment:
Hmm , moim zdaniem Cart powinien przechowywać listę obiektów typu Product. Dzieki temu będziemy mogli nawiązać relacje z encją Product, i przy każdorazowaym dodaniu czegoś do listy w klasie Cart , będzie dodawało się do bazy danych z odpowiednią zależnoscią.  Wydaje mi się że będzie to lepsze wydajnościowo ponieważ wszystkim zajmie się hibernate.  Jak to widze:
- zmiana przechowywanych obiektów z ProductDto na Product
-  nawiązanie relacji pomiędzy encjami Cart oraz product 1:N

Reply:
Long userId  - zamiana na User user   oraz utworzenie relacji 1:1

Reply:
> 
> 
> Wydaje mi się że powinniśmy zmienić:
> 
>     * Long userId  na User user
>       chyba będzie potrzebna tutaj relacja 1:1  co wy na to?

Według mnie tak powinniśmy zrobić. Cart powinnien być tworzony w momencie stworzenia Usera i potem wykonujemy na nim tylko operacji dodawania/usuwania Product, a w momencie złożenia zamówienie pobieramy liste produktów, a potem ją czyścimy. Tak mi się wydaje.

Reply:
Racja, relacje dodane - dzięki!

Zmieniłam też Long userId na User user według wskazówek, ale mam jeszcze co do tego pytanie. Dlaczego w tym wypadku to rozwiązanie jest bardziej trafne od Long userId?

Reply:
Aby zrobić relacje  między encjami  obie klasy muszę mieć dodane pole typu tej drugiej klasy z która zachodzi relacja. 

Chyba że relacja jest jednostronna to wystarczy że jedna klasa będzie posiadała takie pole.   
Ale  wtedy dojdzie do sytuacji że user będzie wiedział że  ma taki ORDER,   ale order nie będzie wiedziało kto jest właścicielem.

Jeżeli mamy relacje wielostronne  Klasa musi mieć kolekcje List, Set która przechowuje obiekty powiązanej encji    Set<Order>  Wtedy user może mieć kilka zamówień.  

Nie stworzymy relacji na polu int poleId  hibernate nie będzie widział z jaką klasą to powiązać.

Pliki DTO służą przenoszeniu informacji gdzieś na inną warstwę komunikacji. Używamy go głównie aby nie wysyłać wszystkich pól np:
Klasa CartDto posiada pole  User user;  Gdy wyślesz takiego JSONA cartDto to wyślesz z Informacją o Cart również cały obiekt user wraz z hasłami, tokenami itp. 
Można by do CarDto   dać pole UserDto  plik userDto już by był ograniczony o pola takie jak hasło  i tokeny. Jednak to też nie jest dobra praktyka.
Jeżeli nie ma takiej potrzeby pliki DTO powinny być proste mieć proste typy pól, najlepiej żeby odzwierciedlał encje.  
W tabeli Cart bazy danych pole User user zapisze się jako kolumna  user_id z referencją do  użytkownika z tabeli User.   Dlatego też w plikach DTO używać tylko userId;  


Zanim zaczniesz przyswajać tą wiedzę niech ktoś potwierdzi że to co mówię jest prawdą ! :)

Reply:
Potwierdzam :) Nie należy mieszać w relacjach obiektów domenowych i obiektów dto. Jako encje robimy tylko obiekty domenowe.